File: csharp/Behavioral/18-Observer/Observer/BillingService.cs
namespace Observer;

public class BillingService : IJobCardObserver
{
    public void Update(string eventName, JobCard jobCard)
    {
        if (eventName == "JobCompleted")
            Console.WriteLine(
                $"  💰 [Billing] Invoice raised for job '{jobCard.Title}' — N${jobCard.Cost:F2}"
            );
    }
}

File: typescript/Behavioral/18-Observer/Observer/BillingService.ts
import type { JobCard } from "../Subject/JobCard";
import { IJobCardObserver } from "./IJobCardObserver";

export class BillingService implements IJobCardObserver {
  update(eventName: string, jobCard: JobCard): void {
    if (eventName === "JobCompleted") {
      console.log(
        `  💰 [Billing] Invoice raised for job '${jobCard.title}' — N$${jobCard.cost.toFixed(2)}`,
      );
    }
  }
}
